packages/nexus/src/client.ts

```diff
@@ -0,0 +1,51 @@
+import { NexusChannel, NexusHandler, NexusWebsocketOptions } from './channel'
+
+export class Nexus {
+  url: string
+
+  constructor(url: string) {
+    if (!url.startsWith('http://') && !url.startsWith('https://')) {
+      throw new Error('Invalid URL, expected http:// or https://')
+    }
+    this.url = url
+  }
+
+  channel(handler: NexusHandler, opts?: NexusWebsocketOptions): NexusChannel {
+    const url = new URL(this.url)
+    url.protocol = url.protocol === 'https:' ? 'wss:' : 'ws:'
+    url.pathname = '/channel'
+    return new NexusChannel(url.toString(), handler, opts)
+  }
+
+  async addRepos(dids: string[]): Promise<void> {
+    const response = await fetch(`${this.url}/add-repos`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ dids }),
+    })
+    await response.arrayBuffer()
+
+    if (!response.ok) {
+      await response.arrayBuffer()
+      throw new Error(`Failed to add repos: ${response.statusText}`)
+    }
+  }
+
+  async removeRepos(dids: string[]): Promise<void> {
+    const response = await fetch(`${this.url}/remove-repos`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ dids }),
+    })
+    await response.arrayBuffer()
```

Comment:
Nit: A more resource efficient way to drain would avoid putting it memory (though in practice it doesn't probably matter)

```ts
    await response.body?.cancel()
```

https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/cancel

> Cancel is used when you've completely finished with the stream and don't need any more data from it, even if there are chunks enqueued waiting to be read.